### PR TITLE
Approver cannot be the request creator

### PR DESCRIPTION
**Pull request by danidoni**
In order to enforce two different reviewers, the approver cannot be the same person than the creator of a request, if the proper attribute is enabled at target project or target package.

## How to test this
1. In the review-app, go to your home project
2. Add an `CreatorCannotAcceptOwnRequests` attribute there.
3. Go to the project's hello-world package
4. Branch it and change the sources so you can submit a request back
5. Go to the request and accept it.

### Expected result
As you have the `CreatorCannotAcceptOwnRequests` attribute set and you are the same approver user as the request's creator user, an error will popup preventing you to approve the request.

If you go now and remove the attribute from the project, you'll be allowed to approve your own request.

You could also approve it with or without the attribute set if you log in with another user with proper permissions in the project (as it's not the same person as the request creator).

## TODO
- [x] Implementation of the validation
- [x] Specs
- [x] Data migration to create the new Attribute
- [x] Testing in API

### CODE REVIEW

File: src/api/app/models/bs_request_permission_check.rb [hunk 1]
```diff
@@ -124,6 +124,13 @@ def cmd_changestate_permissions(opts)
 
       check_newstate_action!(action, opts)
 
+      # TODO: Get the relevant project attribute, from the target project or target package. Retrieve the accepter and check if it's the same person than the creator. And fail if true
+      target_package = Package.get_by_project_and_name(action.target_project, action.target_package)
```
Comment (coolo) [inline src/api/app/models/bs_request_permission_check.rb:128]: note that target packages often don't exist. this code looks crashing in such a case

Comment (danidoni) [reply src/api/app/models/bs_request_permission_check.rb:128]: Yeah, I noticed... Incoming push...

File: src/api/spec/models/bs_request_spec.rb [hunk 2]
```diff
@@ -220,6 +220,128 @@
...
+          end
+
+          it 'accepts the request' do
+            request.change_state(newstate: 'accepted', force: true)
+            expect(request.state).to be(:accepted)
+          end
+        end
+      end
+
+      context 'and the target project does not have an attribute to disallow acceptance by the creator' do
```
Comment (hennevogel) [inline src/api/spec/models/bs_request_spec.rb:272]: This is already tested elsewhere I bet...

File: src/api/spec/models/bs_request_spec.rb [hunk 3]
```diff
@@ -220,6 +220,128 @@
...
+
+          it 'accepts the request' do
+            request.change_state(newstate: 'accepted', force: true)
+
+            expect(request.state).to be(:accepted)
+          end
+        end
+      end
+
+      context 'and the target package does not have an attribute to disallow acceptance by the creator' do
```
Comment (hennevogel) [inline src/api/spec/models/bs_request_spec.rb:327]: Same here. CI time is developer time 💰

**Review by hennevogel — APPROVED** [review]

File: src/api/spec/models/bs_request_spec.rb (hunk 2 above)
Comment (danidoni) [reply src/api/spec/models/bs_request_spec.rb:272]: Hm, okay, I'll remove those specs

**Pull request merged**